src/curl.c

```diff
@@ -331,14 +331,9 @@ int swupd_curl_get_file(const char *url, char *filename, struct file *file,
 		// normal file download
 		struct stat stat;
 
-		if (file) {
-			local = file;
-			local->fh = NULL;
-		} else {
-			local = calloc(1, sizeof(struct file));
-			if (!local) {
-				abort();
-			}
+		local = calloc(1, sizeof(struct file));
```

Comment:
If we don't have an incoming struct file*, then do we need to heap allocate local at all, or can we just stack allocate it? One less potential call to abort() is always a good change in my book.

Reply:
Ah, this happens in the next patch in the set. :grin: